spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkCompressionUtil.java

```diff
@@ -0,0 +1,130 @@
+      return rowBasedCompression(shuffleCodec(spark), outputCodec);
+    } else {
+      return 1.0;
+    }
+  }
+
+  private static String shuffleCodec(SparkSession spark) {
+    SparkConf sparkConf = spark.sparkContext().conf();
+    return shuffleCompressionEnabled(sparkConf) ? sparkCodec(sparkConf) : "none";
+  }
+
+  private static boolean shuffleCompressionEnabled(SparkConf sparkConf) {
+    return sparkConf.getBoolean(SHUFFLE_COMPRESSION_ENABLED, SHUFFLE_COMPRESSION_ENABLED_DEFAULT);
+  }
+
+  private static String sparkCodec(SparkConf sparkConf) {
+    return sparkConf.get(SPARK_COMPRESSION_CODEC, SPARK_COMPRESSION_CODEC_DEFAULT);
+  }
+
+  private static double columnarCompression(String shuffleCodec, String outputCodec) {
+    Pair<String, String> key = Pair.of(normalize(shuffleCodec), normalize(outputCodec));
+    return COLUMNAR_COMPRESSIONS.getOrDefault(key, DEFAULT_COLUMNAR_COMPRESSION);
+  }
+
+  private static double rowBasedCompression(String shuffleCodec, String outputCodec) {
+    Pair<String, String> key = Pair.of(normalize(shuffleCodec), normalize(outputCodec));
+    return ROW_BASED_COMPRESSIONS.getOrDefault(key, DEFAULT_ROW_BASED_COMPRESSION);
+  }
+
+  private static String normalize(String value) {
+    return value != null ? value.toLowerCase(Locale.ROOT) : null;
+  }
+
+  private static Map<Pair<String, String>, Double> initColumnarCompressions() {
+    Map<Pair<String, String>, Double> compressions = Maps.newHashMap();
+
+    compressions.put(Pair.of("none", "zstd"), 4.0);
+    compressions.put(Pair.of("none", "gzip"), 4.0);
+    compressions.put(Pair.of("none", "zlib"), 4.0);
+    compressions.put(Pair.of("none", "snappy"), 3.0);
+    compressions.put(Pair.of("none", "lz4"), 3.0);
+
+    compressions.put(Pair.of("zstd", "zstd"), 2.0);
+    compressions.put(Pair.of("zstd", "gzip"), 2.0);
+    compressions.put(Pair.of("zstd", "zlib"), 2.0);
+    compressions.put(Pair.of("zstd", "snappy"), 1.5);
+    compressions.put(Pair.of("zstd", "lz4"), 1.5);
+
+    compressions.put(Pair.of("snappy", "zstd"), 3.0);
+    compressions.put(Pair.of("snappy", "gzip"), 3.0);
+    compressions.put(Pair.of("snappy", "zlib"), 3.0);
+    compressions.put(Pair.of("snappy", "snappy"), 2.0);
+    compressions.put(Pair.of("snappy", "lz4"), 2.);
+
+    compressions.put(Pair.of("lz4", "zstd"), 3.0);
+    compressions.put(Pair.of("lz4", "gzip"), 3.0);
+    compressions.put(Pair.of("lz4", "zlib"), 3.0);
+    compressions.put(Pair.of("lz4", "snappy"), 2.0);
+    compressions.put(Pair.of("lz4", "lz4"), 2.0);
+
+    return compressions;
+  }
+
+  private static Map<Pair<String, String>, Double> initRowBasedCompressions() {
+    Map<Pair<String, String>, Double> compressions = Maps.newHashMap();
+
+    compressions.put(Pair.of("none", "zstd"), 2.0);
+    compressions.put(Pair.of("none", "gzip"), 2.0);
+
+    compressions.put(Pair.of("lz4", "zstd"), 1.5);
+    compressions.put(Pair.of("lz4", "gzip"), 1.5);
```

Comment:
Why no `zstd` or `snappy` shuffle compression options? Just assume that this will use the default?

Reply:
I should probably add these too, I missed them when I added values for ORC and Parquet.